src/odemis/gui/cont/streams.py

```diff
@@ -1371,47 +1371,13 @@ def is_compatible(c):
 
     def _add_fastem_ctrls(self):
         self.stream_panel.add_divider()
+        # Create the immersion mode button
+        btn_immersion_mode = self.stream_panel.add_immersion_mode_button()
 
-        # FIXME uncomment autostigmation, when the functionality is working correctly.
-        _, btn_autofocus = self.stream_panel.add_run_btn("Autofocus")
-        _, btn_autobc = self.stream_panel.add_run_btn("Auto-brightness/contrast")
-        # _, btn_autostigmation = self.stream_panel.add_run_btn("Autostigmation")
-
-        btn_autofocus.Bind(wx.EVT_BUTTON, self._on_btn_autofocus)
-        btn_autobc.Bind(wx.EVT_BUTTON, self._on_btn_autobc)
-        # btn_autostigmation.Bind(wx.EVT_BUTTON, self._on_btn_autostigmation)
-
-    @call_in_wx_main
-    def _on_btn_autofocus(self, _):
-        self.stream_panel.Enable(False)
-        self.pause()
-        self.pauseStream()
-        f = self.stream.focuser.applyAutofocus(self.stream.detector)
-        f.add_done_callback(self._on_autofunction_done)
-
-    @call_in_wx_main
-    def _on_btn_autobc(self, _):
-        self.stream_panel.Enable(False)
-        self.pause()
-        self.pauseStream()
-        f = self.stream.detector.applyAutoContrastBrightness()
-        f.add_done_callback(self._on_autofunction_done)
-
-    @call_in_wx_main
-    def _on_btn_autostigmation(self, _):
-        self.stream_panel.Enable(False)
-        self.pause()
-        self.pauseStream()
-        f = self.stream.emitter.applyAutoStigmator(self.stream.detector)
-        f.add_done_callback(self._on_autofunction_done)
-
-    @call_in_wx_main
-    def _on_autofunction_done(self, f):
-        self.stream_panel.Enable(True)
-        self.resume()
-        # Don't automatically resume stream, autofunctions can take a long time.
-        # The user might not be at the system after the functions complete, so the stream
-        # would play idly.
+        # Store a setting entry for the immersion mode button
+        se = SettingEntry(name="immersion_mode", va=self.stream.emitter.immersion,
```

Comment:
So this is just for live stream? But as soon as the actual overview acquisition starts, the immersion mode is force on. Is that really how it should behave? Wouldn't it be less confusing for the user if that setting was used during the overview too?

Reply:
Good catch, I didn't know this. It is not the expected behavior. So then I should make is of a `VigilantAttributeConnector` right @pieleric ?

Reply:
I'm not sure I completely understand both your comments, so here is the explanation of the desired behavior:
During live view we want the user to be able to switch between immersion and non-immersion mode, to have full control of the stream. During the overview imaging the immersion mode should always be set to False (this allows us to have a larger field of view). At the end of the overview imaging the immersion mode should be set back to whatever is selected in the GUI.

Reply:
Ok, if it's the expected behaviour, that's at least good code-wise (I don't know if it's obvious to the user). FYI, acq.fastem.OverviewAcquisition automatically disables the immersion mode, via `fastem_conf.configure_scanner()`.

It also automatically enables the immersion mode back on afterwards. It might be something you will want to change (ex, store the current setting and restore it at the end of the overview... but maybe it's not very efficient if multiple scintillators are acquired at once).